back/engines/free/email_campaigns/spec/models/email_campaigns/campaign_spec.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe EmailCampaigns::Campaign, type: :model do
+  describe '#apply_recipient_filters' do
+    let(:campaign) { create(:invite_received_campaign) }
+    let(:invite) { create(:invite) }
+    let(:activity) { create(:activity, item: invite, action: 'created', user: invite.inviter) }
+
+    it 'excludes users with pending invite and no email' do
+      invitee = create(:user, invite_status: 'pending', email: nil, first_name: 'test_name')
+      invite2 = create(:invite, invitee_id: invitee.id)
+      activity2 = create(:activity, item: invite2, action: 'created', user: invite2.inviter)
+
+      result = campaign.apply_recipient_filters(activity: activity2)
+
+      expect(result.to_json).not_to include invitee.first_name
```

Comment:
`result` is a scope, so maybe you can (if you want) test it like that
```suggestion
      expect(result).not_to include invitee
```